src/analysis/calculus/parametric_integral.lean

```diff
@@ -0,0 +1,241 @@
+/-
+Copyright (c) 2021 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import measure_theory.set_integral
+import analysis.calculus.mean_value
+
+/-!
+# Derivatives of integrals depending on parameters
+
+A parametric integral is a function with shape `f = λ x : H, ∫ a : α, F x a ∂μ` for some
+`F : H → α → E`, where `H` and `E` are normed spaces and `α` is a measured space with measure `μ`.
+
+We already know from `continuous_of_dominated` in `measure_theory.bochner_integral` how to
+guarantee that `f` is continuous using the dominated convergence theorem. In this file,
+we want to express the derivative of `f` as the integral of the derivative of `F` with respect
+to `x`.
+
+
+## Main results
+
+As explained above, all results express the derivative of a parametric integral as the integral of
+a derivative. The variations come from the assumptions and from the different ways of expressing
+derivative, especially Fréchet derivatives vs elementary derivative of function of one real
+variable.
+
+* `has_fderiv_at_of_dominated_loc_of_lip`: this version assumes
+    `F x` is ae-measurable for x near `x₀`, `F x₀` is integrable,
+    `λ x, F x a` has derivative `F' a : H →L[ℝ] E` at `x₀` which is ae-measurable,
+    `λ x, F x a` is locally Lipschitz near `x₀` for almost every `a`, with a Lipschitz bound which
+    is integrable with respect to `a`. A subtle point is that the "near x₀" in the last condition
+    has to be uniform in `a`. This is controlled by a positive number `ε`.
+
+* `has_fderiv_at_of_dominated_of_fderiv_le`: this version assume `λ x, F x a` has derivative
+    `F' x a` for `x` near `x₀` and `F' x` is bounded by an integrable function independent from
+    `x` near `x₀`.
+
+
+`has_deriv_at_of_dominated_loc_of_lip` and `has_deriv_at_of_dominated_loc_of_deriv_le ` are versions
+of the above two results that assume `H = ℝ` and uses the high-school derivative `deriv` instead of
+Fréchet derivative `fderiv`.
+-/
+
+noncomputable theory
+
+open topological_space measure_theory filter metric
+open_locale topological_space filter
+
+variables {α : Type*} [measurable_space α] {μ : measure α}
+          {E : Type*} [normed_group E] [normed_space ℝ E]
+          [complete_space E] [second_countable_topology E]
+          [measurable_space E] [borel_space E]
+          {H : Type*} [normed_group H] [normed_space ℝ H]
+          [second_countable_topology $ H →L[ℝ] E]
+
+lemma has_fderiv_at_of_dominated_loc_of_lip' {F : H → α → E} {F' : α → (H →L[ℝ] E)} {x₀ : H}
```

Comment:
Could you add a docstring to all lemmas in this file, not repeating all the assumptions but explaining their flavor and the flavor of the conclusion, and highlighting the differences with the other versions (and giving pointers to other versions when relevant)?

Reply:
I tried to add docstrings, but telling the story of each lemma in a single sentence does not make a fun reading. I could have used lists in docstrings but then I don't much difference with reading the actual statements. Feel free to directly modify those docstrings if you think you can do better.